src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -4372,15 +4474,21 @@ private ImmutableArray<TypeWithAnnotations> InferMethodTypeArguments(
             // a possibly constructed generic type, is exceedingly subtle. See the comments
             // in "Infer" for details.
 
+            PooledDictionary<TypeParameterSymbol, int> ordinals = makeOrdinalsIfNeeded(member, originalTypeParameters);
+            NamedTypeSymbol containingType = member.GetIsNewExtensionMember() ? member.ContainingType.ContainingType : member.ContainingType;
```

Comment:
>containingType [](http://example.com/codeflow?start=28&length=14)

"containingTypeForInference"? This is a value that is valid to use only in one specific setting. Correct? It would be good to make it clear so that the local was not reused by mistake in other context just because its name. At the same time, it doesn't look useful to have this local at all.